### PR TITLE
remove ioutil and errors, update dependencies

### PR DESCRIPTION
**Pull request by majst01**
- replace ioutil with newer funcs
- replace old errors
- update dependencies
- test new metal-stack/security

TODO:
- [x] test in test environment

### CODE REVIEW

File: cmd/metal-api/internal/metal/errors.go [hunk 1]
```diff
@@ -13,30 +15,68 @@ var (
...
 // IsInternal checks if an error is a Internal error.
 func IsInternal(e error) bool {
-	return errors.Is(errors.Cause(e), errInternal)
+	return errors.Is(e, errInternal)
+}
+
+// wrapf returns an error annotating err with a stack trace
+// at the point Wrapf is called, and the format specifier.
+// If err is nil, Wrapf returns nil.
+func wrapf(err error, format string, args ...interface{}) error {
```
Comment (Gerrit91) [inline cmd/metal-api/internal/metal/errors.go:49]: This is the only part which makes me hesitate. Function is obviously taken from https://github.com/pkg/errors/blob/master/errors.go#L201, but is it really an improvement to copy it?

Comment (majst01) [reply cmd/metal-api/internal/metal/errors.go:49]: Agreed, i will revert this particular change, test are failing because of this i guess

Comment (majst01) [reply cmd/metal-api/internal/metal/errors.go]: reverted

**Comment by Gerrit91** [comment]
Maybe needs protoc generation with 1.27.1? 

https://github.com/metal-stack/builder/pull/27

**Review by Gerrit91 — APPROVED** [review]

**Comment by majst01** [comment]
i keep it running in test for a day or two before merging

**Comment by majst01** [comment]
already included in #202

**Pull request closed without merging**